karyon3-rxnetty/src/main/java/com/netflix/karyon/rxnetty/HttpEndpointDefinition.java

```diff
@@ -0,0 +1,66 @@
+package com.netflix.karyon.rxnetty;
+
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.HttpServerRequest;
+import io.reactivex.netty.protocol.http.server.HttpServerResponse;
+import io.reactivex.netty.protocol.http.server.RequestHandler;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.inject.Inject;
+
+import rx.Observable;
+
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Scopes;
+import com.netflix.karyon.http.UriPatternMatcher;
+
+public class HttpEndpointDefinition {
+    private final String pattern;
+    private final Key<? extends RequestHandler<ByteBuf, ByteBuf>> endpointKey;
+    private final UriPatternMatcher patternMatcher;
+    private final AtomicReference<RequestHandler<ByteBuf, ByteBuf>> httpEndpoint = new AtomicReference<RequestHandler<ByteBuf, ByteBuf>>();
+
+    public HttpEndpointDefinition(
+            String pattern,
+            Key<? extends RequestHandler<ByteBuf, ByteBuf>> endpointKey,
+            UriPatternMatcher patternMatcher,
+            RequestHandler<ByteBuf, ByteBuf> endpointInstance) {
+        this.pattern = pattern;
+        this.endpointKey = endpointKey;
+        this.patternMatcher = patternMatcher;
+        this.httpEndpoint.set(endpointInstance);
+    }
+
+    boolean shouldServe(String uri) {
+        return uri != null && patternMatcher.matches(uri);
```

Comment:
If it can be null, then the user might want it to match, e.g., a default page redirect endpoint. Right now there is no way to specify a RequestHandler that would get used when it is null.